sefaria/helper/topic.py

```diff
@@ -1037,6 +1039,39 @@ def topic_change_category(topic_obj, new_category, old_category="", rebuild=Fals
         rebuild_topic_toc(topic_obj, category_changed=True)
     return topic_obj
 
+def update_topic_titles(topic_obj, data):
+    for lang in ['en', 'he']:
+        for title in topic_obj.get_titles(lang):
+            topic_obj.remove_title(title, lang)
+        for title in data['altTitles'][lang]:
+            topic_obj.add_title(title, lang)
+
+    topic_obj.add_title(data['title'], 'en', True, True)
+    topic_obj.add_title(data['heTitle'], 'he', True, True)
+    return topic_obj
+
+
+def update_authors_data(topic_obj, data, dataSource='learning-team-editing-tool'):
+    # update place info added to author, then update year and era info
+    if not hasattr(topic_obj, 'properties'):
+        topic_obj.properties = {}
+    process_topic_place_change(topic_obj, data)
+    topic_obj = update_author_era(topic_obj, data, dataSource='learning-team-editing-tool')
+
+    return topic_obj
+
+def update_author_era(topic_obj, data, dataSource='learning-team-editing-tool'):
+    for k in ["birthYear", "deathYear"]:
+        year = data.get(k, '')  # set to None if blank, otherwise validate number
+        if year != '':
+            assert int(year), f"'{k}' must be a number but is {year}"
```

Comment:
the year being an integer should be validated at the API level even before this function and return a 400 with an explanation of the error. That being said, this is usually done in a cleaner way which we are not set up for that. So best to put in a comment alongside a type check at the API level